src/FileLogger.java

```diff
@@ -0,0 +1,105 @@
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class FileLogger {
+
+    private FileLoggerConfiguration config;
+    private File logFile;
+    private long fileSize;
+
+    public File getLogFile() {
+        return logFile;
+    }
+    public void setLogFile(File logFile) {
+        this.logFile = logFile;
+    }
+
+    public FileLogger(FileLoggerConfiguration config) {
+        this.config = config;
+        this.logFile = new File(config.getFilePath());
+        if (!logFile.exists()) {
+            try {
+                logFile.createNewFile();
```

Comment:
You should not create a log file in the constructor, it should be created when you're writing some logs into that file. Remove it.